sintr_analysis_server_example/lib/query/severe_log.dart

```diff
@@ -89,17 +51,44 @@ class SevereLogMapper extends _AbstractFailureMapper {
   void _processSevereLogMsg(int time, String logMessageText) {
     var start = logMessageText.indexOf(':', 7);
     var msgText = logMessageText.substring(start + 1).replaceAll('::', ':');
-
-    // Record all except getContent failed messages
-    if (!msgText.startsWith(GET_CONTENTS_FAILED_PREFIX)) {
-      _recordFailure(time, SEVERE_LOG, msgText);
-      return;
-    }
+    _recordFailure(time, SEVERE_LOG, msgText);
   }
 }
 
 abstract class _AbstractFailureMapper extends InstrumentationMapper {
-  void _recordFailure(int time, String resultType, [resultData]) {
-    addResult(sdkVersion, [sessionId, time, resultType, resultData]);
+  static const MAX_TIMES_TO_REPORT = 1000 * 1000;
+
+  // type -> data -> times
+  Map<String, Map<String, List<int>>> typeResultsTimes = {};
+  Map<String, Map<String, int>> typeResultsCounts = {};
+
+  void _recordFailure(int time, String resultType, String resultData) {
+    String data = resultData.split('\n')[0];
+
+    typeResultsTimes.putIfAbsent(resultType, () => {});
+    typeResultsCounts.putIfAbsent(resultType, () => {});
+
+    Map<String, List<int>> resultTimes = typeResultsTimes[resultType];
+    Map<String, int> resultCounts = typeResultsCounts[resultType];
```

Comment:
Both of these can be combined with the putIfAbsent lines above because [putIfAbsent returns the value](https://api.dartlang.org/1.13.0/dart-core/Map/putIfAbsent.html).
```
Map<String, List<int>> resultTimes = typeResultsTimes.putIfAbsent(resultType, () => {});
```